pad.go

```diff
@@ -80,20 +81,23 @@ func (pad *PAD) Update(policies Policies) {
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	index := pad.computePrivateIndex(key)
+	return pad.tree.Set(index, key, value)
 }
 
 func (pad *PAD) Lookup(key string) *AuthenticationPath {
 	str := pad.currentSTR
-	return str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
+	return str.tree.Get(lookupIndex)
 }
 
 func (pad *PAD) LookupInEpoch(key string, epoch uint64) (*AuthenticationPath, error) {
 	str := pad.GetSTR(epoch)
 	if str == nil {
 		return nil, ErrorSTRNotFound
 	}
-	ap := str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
```

Comment:
Look at the implementation of `computePrivateIndex`,

```
vrfPivKey := pad.currentSTR.policies.vrfPrivate()
```

meaning that it gets the private key from the current epoch.

But this is a lookup in a specific epoch (the `str := pad.GetSTR(epoch)`)

So, it seems like it needs to be,
```
str.policies.vrfPrivate()
```

And I'm assuming that the private key change in the policy means we need to recompute all the indices.